warehouse/query-core/src/main/java/datawave/query/iterator/logic/WordsAndScores.java

```diff
@@ -0,0 +1,101 @@
+package datawave.query.iterator.logic;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+/**
+ * An object used to save terms and their respective scores (if a score exists and is valid).
+ */
+public class WordsAndScores {
+    /** the arraylist of words */
+    private final ArrayList<String> words;
+    /** the arraylist of scores */
+    private final ArrayList<Integer> scores;
```

Comment:
One possible approach in this class is to make a WordScorePair supporting class. This would allow you to have one array list here so you wouldn't have to coordinate between two arrays. Would also allow you to provide default values upon construction.

Reply:
good idea. might implement